# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/ContentBlocking.cpp
# Commit: bdec9be4313a
# Full Hash: bdec9be4313a86a6bf62ab896188b641e10befae
# Author: Dimi Lee <dlee@mozilla.com>
# Date: 2020-05-01 15:43:38
# Description:
#   Bug 1634414 - Check principal before using IsThirdPartyPrincipal in AllowAccessFor r=timhuang
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73418
# ==============================================================================

diff -r 697891f0e71a -r bdec9be4313a toolkit/components/antitracking/ContentBlocking.cpp
--- a/toolkit/components/antitracking/ContentBlocking.cpp	Thu Apr 30 15:43:16 2020 +0000
+++ b/toolkit/components/antitracking/ContentBlocking.cpp	Fri May 01 09:05:17 2020 +0000
@@ -322,6 +322,10 @@
       bool isThirdParty;
       nsCOMPtr<nsIPrincipal> principal =
           AntiTrackingUtils::GetPrincipal(aParentContext);
+      if (!principal) {
+        LOG(("Can't get the principal from the browsing context"));
+        return StorageAccessGrantPromise::CreateAndReject(false, __func__);
+      }
       Unused << trackingPrincipal->IsThirdPartyPrincipal(principal,
                                                          &isThirdParty);
       runInSameProcess = !isThirdParty;
